src/App/integrationTests/App.addSampleUnitRouting.test.js

```diff
@@ -14,23 +14,26 @@ import { getMockDexieInstancesAllSuccess } from '../../testUtilities/mockDexie'
 test('Clicking Add Sample Unit then click Fish Belt link expects to see New Fish Belt page.', async () => {
   const { dexiePerUserDataInstance, dexieCurrentUserInstance } = getMockDexieInstancesAllSuccess()
 
-  renderAuthenticatedOnline(<App dexieCurrentUserInstance={dexieCurrentUserInstance} />, {
-    initialEntries: ['/projects/5/collecting'],
-    dexiePerUserDataInstance,
-    dexieCurrentUserInstance,
-  })
+  const { user } = renderAuthenticatedOnline(
+    <App dexieCurrentUserInstance={dexieCurrentUserInstance} />,
+    {
+      initialEntries: ['/projects/5/collecting'],
+      dexiePerUserDataInstance,
+      dexieCurrentUserInstance,
+    },
+  )
 
   await screen.findByLabelText('project pages loading indicator')
   await waitForElementToBeRemoved(() => screen.queryByLabelText('project pages loading indicator'))
 
-  userEvent.click(
+  await user.click(
```

Comment:
with the upgrades we now need to await user events in tests. Sometimes we even will need to assert that the event is complete (eg form input. Await the input, then assert that the expected value is there)

Reply:
I also found that regular assertions with `await` and `findBy*` needed to be converted to use `await waitFor`